auto_dev/services/publish/index.py

```diff
@@ -0,0 +1,171 @@
+"""This module contains the service logic for publishing agents."""
+
+import shutil
+from typing import Optional
+from pathlib import Path
+
+from aea.configurations.base import PublicId
+from aea.configurations.data_types import PackageType
+
+from auto_dev.enums import FileType, LockType
+from auto_dev.utils import change_dir, get_logger, load_autonolas_yaml
+from auto_dev.exceptions import OperationError
+from auto_dev.cli_executor import CommandExecutor
+
+
+logger = get_logger()
+
+
+class PublishService:
+    """Service for publishing agents and managing packages."""
+
+    def __init__(self, verbose: bool = False):
+        """Initialize the publish service.
+
+        Args:
+            verbose: whether to show verbose output.
+        """
+        self.verbose = verbose
+
+    def ensure_local_registry(self) -> None:
+        """Ensure a local registry exists.
+
+        Raises:
+            OperationError: if the command fails.
+        """
+        # If we're in an agent directory, packages should be in parent dir
+        packages_dir = Path("../packages") if Path("aea-config.yaml").exists() else Path("packages")
+        if not packages_dir.exists():
+            # Need to cd to parent dir if initializing there
+            if packages_dir.parent != Path("."):
+                with change_dir(".."):
+                    command = CommandExecutor(["poetry", "run", "autonomy", "packages", "init"])
+                    result = command.execute(verbose=self.verbose)
+                    if not result:
+                        msg = f"Command failed: {command.command}"
+                        raise OperationError(msg)
+            else:
+                command = CommandExecutor(["poetry", "run", "autonomy", "packages", "init"])
+                result = command.execute(verbose=self.verbose)
+                if not result:
+                    msg = f"Command failed: {command.command}"
+                    raise OperationError(msg)
```

Comment:
not sure about all this, 
Its a bit unstruictired and as were doing file level modification, might be better to for example, 

A) you run the init command no matter where you are,

B) you are actually checking 2 things.

ii) is the packages initialised,
iii) what directory are you in.


Do the directory check get a package_path out and then use a context manager to switch to that directory prior to the execution of the command